pkgs/applications/networking/cluster/terraform-providers/update-all-providers

```diff
@@ -5,18 +5,21 @@
 # Update all providers which have specified provider source address
 set -euo pipefail
 
-providers=$(
+readarray -t providers < <(
   jq -r 'to_entries
   | map_values(.value + { alias: .key })
   | .[]
   | select(."provider-source-address"?)
   | .alias' providers.json
 )
 
-echo "Will update providers:"
-echo "${providers}"
+cat <<EOF
+Will update ${#providers[@]} providers:
 
-for provider in ${providers}; do
-  echo "Updating ${provider}"
+${providers[@]}
```

Comment:
nit: use `${providers[*]}` when it gets splatted in a string. You can also set `IFS='\n'` to print them each on a separate line.

Reply:
What is the semantic difference?

Reply:
Changed to use `*`. Not using separate lines is intentional as ~135 providers is a bit spammy in a terminal.

Reply:
@Mic92 @ is to "splat" an array into individual arguments. To see the difference, https://replit.com/@zimbatm/splat-args?v=1#main.sh